specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json

```diff
@@ -0,0 +1,2590 @@
+      "required": [
+        "schedule",
+        "name"
+      ],
+      "type": "object",
+      "properties": {
+        "schedule": {
+          "description": "The CRON expression for the task schedule",
+          "type": "string"
+        },
+        "status": {
+          "description": "The current status of trigger.",
+          "default": "Enabled",
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "TriggerStatus",
+            "modelAsString": true
+          }
+        },
+        "name": {
+          "description": "The name of the trigger.",
+          "type": "string"
+        }
+      }
+    },
+    "SourceTrigger": {
+      "description": "The properties of a source based trigger.",
+      "required": [
+        "sourceRepository",
+        "sourceTriggerEvents",
+        "name"
+      ],
+      "type": "object",
+      "properties": {
+        "sourceRepository": {
+          "$ref": "#/definitions/SourceProperties",
+          "description": "The properties that describes the source(code) for the task."
+        },
+        "sourceTriggerEvents": {
+          "description": "The source event corresponding to the trigger.",
+          "type": "array",
+          "items": {
+            "enum": [
+              "commit",
+              "pullrequest"
+            ],
+            "type": "string",
+            "x-ms-enum": {
+              "name": "SourceTriggerEvent",
+              "modelAsString": true
+            }
+          }
+        },
+        "status": {
+          "description": "The current status of trigger.",
+          "default": "Enabled",
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "TriggerStatus",
+            "modelAsString": true
+          }
+        },
+        "name": {
+          "description": "The name of the trigger.",
+          "type": "string"
+        }
+      }
+    },
+    "BaseImageTrigger": {
+      "description": "The trigger based on base image dependency.",
+      "required": [
+        "baseImageTriggerType",
+        "name"
+      ],
+      "type": "object",
+      "properties": {
+        "baseImageTriggerType": {
+          "description": "The type of the auto trigger for base image dependency updates.",
+          "enum": [
+            "All",
+            "Runtime"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BaseImageTriggerType",
+            "modelAsString": true
+          }
+        },
+        "updateTriggerEndpoint": {
+          "description": "The endpoint URL for receiving update triggers.",
+          "type": "string"
+        },
+        "includeTriggerMetadata": {
+          "description": "Include Trigger metadata on Base image update triggers.",
+          "default": true,
+          "type": "boolean"
```

Comment:
We should avoid booleans in swagger, we typically use string enums as they're extensible and forward compatible.
Consider something more along the lines of an optional property that looks like this:
'updateTriggerContent' : ['triggerMetadata']

Reply:
done, converted to an enum 'updateTriggerPayloadType' as seen in the new swagger